NumLib/DOF/DOFTableUtil.cpp

```diff
@@ -30,4 +44,60 @@ std::vector<GlobalIndexType> getIndices(
     return indices;
 }
 
+double norm(GlobalVector const& x, unsigned const global_component,
+            MathLib::VecNormType norm_type,
+            LocalToGlobalIndexMap const& dof_table, MeshLib::Mesh const& mesh)
+{
+    // TODO that also includes ghost nodes.
+    std::size_t const n_nodes = mesh.getNumberOfNodes();
+
+    // TODO for PETSc some global accumulation is necessary.
+    switch(norm_type)
+    {
+    case MathLib::VecNormType::NORM1:
```

Comment:
Maybe you could implement `norm1`, `norm2` and the infinity norm in separate functions and use `norm` as a wrapper. This way it is possible to "hard code" the norm function if this makes sense. In other situations one can use your flexible variant.